actor/builtin/miner/miner_test.go

```diff
@@ -514,6 +548,40 @@ func TestGetProofsMode(t *testing.T) {
 	})
 }
 
-func parseAbiBoolean(bytes []byte) bool {
-	return bytes[0] == 1
+
+
+func TestLatePoStFee(t *testing.T) {
+	pledgeCollateral := af(1000)
+
+	t.Run("on time charges no fee", func(t *testing.T) {
+		assert.True(t, types.ZeroAttoFIL.Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(999), bh(100))))
+		assert.True(t, types.ZeroAttoFIL.Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1000), bh(100))))
+	})
+
+	t.Run("fee proportional to lateness", func(t *testing.T) {
+		// 1 block late is 1% of 100 allowable
+		assert.True(t, af(10).Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1001), bh(100))))
+		// 5 blocks late of 100 allowable
+		assert.True(t, af(50).Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1005), bh(100))))
+
+		// 2 blocks late of 10000 allowable, fee rounds down to zero
+		assert.True(t, af(0).Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1002), bh(10000))))
+		// 9 blocks late of 10000 allowable, fee rounds down to zero
+		assert.True(t, af(0).Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1009), bh(10000))))
+		// 10 blocks late of 10000 allowable is 1/1000
+		assert.True(t, af(1).Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1010), bh(10000))))
+	})
+
+	t.Run("fee capped at total pledge", func(t *testing.T) {
+		assert.True(t, pledgeCollateral.Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1100), bh(100))))
+		assert.True(t, pledgeCollateral.Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(2000), bh(100))))
+	})
+}
+
+func af(h int64) types.AttoFIL {
```

Comment:
I like these.  Wonder if they belong somewhere more generally accessible?

Reply:
I don't think so. If widely available, would need to have better names and descriptions which would defeat the purpose. It's totally acceptable for this to be duplicated a few times.